anta/custom_types.py

```diff
@@ -17,13 +18,40 @@ def aaa_group_prefix(v: str) -> str:
     return f"group {v}" if v not in built_in_methods and not v.startswith("group ") else v
 
 
+def interface_autocomplete(v: str) -> str:
+    """Allow the user to only provide the beginning of an interface name
+    E.g. et, eth will be changed to Ethernet"""
+    intf_id_re = re.compile(r"[0-9]+(\/[0-9]+)*(\.[0-9]+)?")
+    m = intf_id_re.search(v)
+    if m is None:
+        raise ValueError(f"Could not parse interface ID in interface '{v}'")
+    intf_id = m[0]
+    if any(v.lower().startswith(p) for p in ["et", "eth"]):
+        return f"Ethernet{intf_id}"
```

Comment:
Should we document this?
Also it only works for ethernet interfaces in the end for now right? Would we not rather force people to have nice catalog inputs rather than taking the load?
(I agree with keep case sensitivity - but for instance what do we do with Port-Channels / Port-channel / Port-ChaNNel?)

Reply:
actually, we support "Port-ChaNNel" now :)
For the rest, I could see this coming so I've implemented it...
But we could support "po", similar as what is done in EOS.

Reply:
added doc